packages/system/src/conditional.test.tsx

```diff
@@ -0,0 +1,31 @@
+import { conditional } from './conditional';
+
+test('returns an array of strings', () => {
+  const variant = conditional('input', { checked: true, disabled: true });
+  expect(variant).toEqual(expect.arrayContaining([expect.any(String)]));
+});
+
+test('returns the variant and the conditional states', () => {
+  const variant = conditional('input', { checked: true, disabled: true });
+  expect(['input', 'input.:checked', 'input.:disabled']).toEqual(
+    expect.arrayContaining(variant)
+  );
+});
+
+test('returns only truthy states', () => {
+  const variant = conditional('input', { checked: true, disabled: false });
+  expect(['input', 'input.:checked']).toEqual(expect.arrayContaining(variant));
+});
+
+test('removes dot from the end of the variant string', () => {
```

Comment:
Isn't that part what box does?

Reply:
Yes, but I need this function directly in conditional() this is why I cannot use it with the Box like others.
One option is to export the ensureCorrectVariant() and use it. But then I have to make [0] because this returns an array.

Reply:
I can make it like this: ` const correctVariant = ensureCorrectVariant(variant)[0];`